### ISSUE TITLE
Need a cancel button for Chat Messages

### ISSUE THREAD

**Issue by IanMayo** [serge-web/serge#1860]
We now temporarily store (cache) the contents of messages that are being edited.

![image](https://user-images.githubusercontent.com/1108513/191449477-5652e571-2c53-4284-8fdc-8055092267be.png)

Now, when we refresh the browser, the Chat Message will keep any draft content.  But, if I change my mind about creating the chat message, I have to manually select-all in the public/private text boxes, and delete them.  I _think_ we would benefit from a `Cancel` button.  One place for this would be beneath (or near) the `Send` right-arrow icon.

Any thoughts @ShendMikullovci ?

**Comment by ShendMikullovci** [comment]
Hi Ian! Yes, a cancel feature does sound beneficial now. I think it would be a little messy to have the button near the Send icon. But we could use an X button instead: 

![Chat X Feature](https://user-images.githubusercontent.com/85491794/191791882-51bbfe8e-63e3-4330-8bdd-d1e78fb8273b.png)

**Comment by IanMayo** [comment]
Sure, ok.  Let's add it to the list of jobs for @lebaphi

**Comment by IanMayo** [comment]
Hello @ShendMikullovci - we have a working version of this here: https://serge-1860-cancel-butto-ua8zpm.herokuapp.com/?wargame=wargame-kskl1i3t&access=pkskl17ti

(see `White Chat` channel`).

In a chat channel - the `new message` component is always present, they don't collapse.  When playing with it, the `X` _feels_ like it should close/collapse the new message panel. Well, that's the standard purpose of the `X` in the corner.   Can you think of an alternate icon/button to use that represents `Clear message`.  Maybe an eraser or delete-left icon?

**Comment by ShendMikullovci** [comment]
> Well, that's the standard purpose of the `X` in the corner.

Good point. The benefit of having the X is that it stands outside of the messag field, allowing for enough functional space for the arrow. In its stead, we could play around with these ideas:

![Eraser Outside](https://user-images.githubusercontent.com/85491794/192540918-62916095-97ab-4dbc-af92-35e97c9c22f5.png)

Eraser icon. Feels somewhat out of place. Like you said, we're used to having buttons like these be closing buttons or minimizers. 

![Delete Left Outside](https://user-images.githubusercontent.com/85491794/192541150-6052e424-c5b9-4abb-9124-a95a2299b493.png)

Delete left also feels a little out of place, but slightly more intuitive than the eraser. Perhaps because it's essentially an X button, but with a tag aimed towards the text. 

![Trash Outside](https://user-images.githubusercontent.com/85491794/192541341-9a2e66cb-ee6d-4366-aa7a-267d0d8b9966.png)

The trash icon is pretty straightforward. I'm not sure how much it insinuates that the message is collapsible though...

If we were to add icons _inside_ of the message box, we'd have to distinguish them from the arrow by color and structure, so the user doesn't accidentally click on one instead of the other. Here they are on the inside: 

![X Inside](https://user-images.githubusercontent.com/85491794/192541660-9692cddc-cc3a-4b8d-8a46-e9ae3445cc4f.png)

Similarly, now the X feels a little out of place, since we're used to seeing it on top...

![Eraser Inside](https://user-images.githubusercontent.com/85491794/192541788-8e84a383-6516-4dd2-a57a-8e54349aa7e6.png)

The eraser looks and feels better than before, but I have this tendency to think it belongs inside the text box instead. Not sure how smart that would be...

![Delete Left Inside](https://user-images.githubusercontent.com/85491794/192541981-a1831ee2-0743-46df-9fb8-f49dcfae7dfe.png)

Pretty straightforward. Common icon aimed towards the text to indicate the impacted area.

![Trash Inside](https://user-images.githubusercontent.com/85491794/192542121-4dc62317-56f7-475d-bd45-4e8571e8dbec.png)

Straightforward as well. Common icon both outside and within SERGE. Makes sense intuitively.

I'm opting for Delete Left and the Trash bin inside and/or outside of the message box. What do you guys think?

**Comment by IanMayo** [comment]
The big solid red trash can looks really clear to me:
![image](https://user-images.githubusercontent.com/1108513/192543276-7b9bc0aa-1df1-48bf-b260-9b4862d5321d.png)

**Comment by ShendMikullovci** [comment]
> The big solid red trash can looks really clear to me: ![image](https://user-images.githubusercontent.com/1108513/192543276-7b9bc0aa-1df1-48bf-b260-9b4862d5321d.png)

If you think it'll look clear to the players, let's go for it.

**Comment by IanMayo** [comment]
Ok @ShendMikullovci - there is a playable version here: https://serge-sql.herokuapp.com/?wargame=wargame-kqgy546g&access=rkrlw6f5f. (look in a chat channel)

**Comment by ShendMikullovci** [comment]
Unfortunately, I came across a UX issue with the design already...

I was trying to test the feature and accidentally hit send instead of delete message. I realized the buttons may be too close together. Furthermore, we're used to the send button being on the right hand side of the message box, so introducing another button on the right hand side automatically makes them compete between one another. I came up with some variations: 

![Trash Bin Variations](https://user-images.githubusercontent.com/85491794/193013389-eb091466-e7b6-4bfd-9064-5627837460ba.png)

Design 1: We display them horizontally instead of vertically stacked upon one another. This respects the consensus of the send button being on the far right while fully distinguishing the delete button as well. 

Design 2: Same horizontal display, except we take them out of the text box. It shrinks the size of the text box a little, but the box remains quite large (especially since the designs here use the minimized window version of the message boxes). What we lose in text box size, we gain in focus: the text box itself becomes unencumbered with anything but its sole purpose, which is typing text.

Design 3: Same horizontal display on the outside of the text box, but without separation. This reintroduces the issue we currently have (competition between buttons) but on a horizontal scale. However, it shrinks the text box less and is consistent with our current designs in other areas of the game (mainly admin pages). 

Design 4: Here, we would be fully distinguishing between message box and text box. With this design, we respect the consensus of the send button being on the right. But it being on the right hand side of the _text box_ is what distinguishes it from the entire _message box_, where we introduce the trash bin. Essentially, this feature is built to delete a _cached text_. But for the user, it serves the purpose of deleting an old _message_. I think this design should work well. 

Design 5: The flip side of design 4. Here, we also distinguish between text and message box, but we display it according to our paradigm: the trash bin symbolizes _the deletion of a cached text_, and the arrow the sending of a message. This design also respects consensus for the arrow being on the right hand side.

Let me know your thoughts.

P.S. Notice I used a new arrow icon for the send button. I think a paper-plane icon is much more intuitive to a message being sent than the 2D arrow.
